lib/orderbook/OrderBook.ts

```diff
@@ -84,10 +90,27 @@ class OrderBook extends EventEmitter {
     return this.currencyInstances.keys();
   }
 
-  constructor(private logger: Logger, models: Models, public nomatching = false,
-    private pool: Pool, private swaps: Swaps, private nosanitychecks = false) {
+  constructor({ logger, models, nomatching, pool, swaps, nosanitychecks }:
+  {
+    logger: Logger,
+    models: Models,
+    nomatching?: boolean,
+    pool: Pool,
+    swaps: Swaps,
+    nosanitychecks?: boolean,
+  }) {
     super();
 
+    this.logger = logger;
+    this.pool = pool;
+    this.swaps = swaps;
+    if (nomatching !== undefined) {
```

Comment:
What do you think about removing this and `nosanitychecks !== undefined` conditional and assigning them a default value when destructuring?

```
...
  constructor(
    { logger, models, pool, swaps, nomatching = false, nosanitychecks = false }:
    {
      logger: Logger,
      models: Models,
      pool: Pool,
      swaps: Swaps,
      nomatching: boolean,
      nosanitychecks: boolean,
    },
  ) {
...
```

Reply:
Up to you!

Reply:
OK I went ahead and changed it back and decided it's a little bit cleaner, going with your approach above.